plugins/Other.js

```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
       })
   );
 
+  fastify.decorate('createAimJsons', filteredOsirixArr => {
+    try {
+      const aimJsons = [];
+      filteredOsirixArr.forEach(el => {
+        const aim = new Aim(el.seedData, fastify.enumAimType.imageAnnotation);
+        const markupsToSave = el.rois.map(roi => fastify.formMarupksToSave(roi));
+        fastify.createAimMarkups(aim, markupsToSave);
+        const aimJson = JSON.parse(aim.getAim());
+        aimJsons.push(aimJson);
+      });
+      return aimJsons;
+    } catch (err) {
+      return new InternalError('Creating aim jsons', err);
+    }
+  });
+
+  fastify.decorate('enumAimType', {
+    imageAnnotation: 1,
+    seriesAnnotation: 2,
+    studyAnnotation: 3,
+  });
+
+  fastify.decorate('formMarupksToSave', roi => {
+    // eslint-disable-next-line camelcase
+    const { SOPInstanceUID, Type, IndexInImage, Max, Mean, Min, Dev, LengthCm, AreaCm2 } = roi;
+    const points = fastify.createPointsArrForOsirix(roi.Point_px);
+    return {
+      imageReferenceUid: SOPInstanceUID,
+      markup: {
+        max: Max,
+        mean: Mean,
+        min: Min,
+        stdDev: Dev,
+        points,
+        LengthCm,
+        AreaCm2,
+      },
+      shapeIndex: IndexInImage,
+      type: Type,
+    };
+  });
+
+  fastify.decorate('createAimMarkups', (aim, markupsToSave) => {
+    // eslint-disable-next-line no-unused-vars
+
+    markupsToSave.forEach(value => {
+      const { type, markup, shapeIndex, imageReferenceUid } = value;
+      // eslint-disable-next-line default-case
+      // eslint-disable-next-line default-case
+      switch (type) {
```

Comment:
shapes ={19: point, 5:line, 14:line,15: polygon...}
supportedtypes array include => lookup
here shapes(type) 
case 'point':...

Reply:
Filed ticket  EPAD-3171 for version 4.2